delivery_purchase_label/models/purchase_order.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import SUPERUSER_ID, _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    delivery_label_picking_id = fields.Many2one(
+        "stock.picking",
+        "Delivery Label Picking",
+        store=True,
+        readonly=True,
+    )
+
+    @api.model
+    def _states_to_generate_delivery_label(self):
+        """Labels will be (re)generated only if the PO is in one of these states."""
+        return ["draft", "sent"]
+
+    @api.model
+    def _mail_templates_to_not_attach_labels(self):
+        return self.env.ref("purchase.email_template_edi_purchase_reminder").ids
+
+    def action_rfq_send(self):
+        self.ensure_one()
+        self._generate_purchase_delivery_label()
+        return super().action_rfq_send()
+
+    def _is_valid_for_vendor_labels(self):
+        self.ensure_one()
+        if self.state not in self._states_to_generate_delivery_label():
+            return
+        if not self.dest_address_id:
+            return False
+        if not any(
+            product.type in ["product", "consu"]
+            for product in self.order_line.product_id
+        ):
+            return False
+        return True
+
+    def _generate_purchase_delivery_label(self):
+        """Create a transfer to generate the carrier labels."""
+        self.ensure_one()
+        if not self._is_valid_for_vendor_labels():
+            return
+        # Find the carrier that will be used
+        carrier = self.partner_id.purchase_delivery_carrier_id
+        if not carrier.purchase_label_picking_type:
+            return
+        if not self.partner_id.property_stock_supplier.id:
+            raise UserError(
+                _(
+                    "You must set a Vendor Location for this partner %s",
+                    self.partner_id.name,
+                )
+            )
+        order = self.with_company(self.company_id)
+        # Create and process the transer to send the labels
+        values = order._get_purchase_delivery_label_picking_value(carrier)
+        picking = self.env["stock.picking"].with_user(SUPERUSER_ID).create(values)
+        moves = order.order_line._create_stock_moves(picking)
+        # Remove the link on the sale and purchase
+        # To not impact the delivered quantity on them
+        picking.sale_id = False
+        moves.sale_line_id = False
+        moves.purchase_line_id = False
+        picking.action_assign()
+        for move in moves:
```

Comment:
The moves can change on action_assign
```suggestion
        picking.action_assign()
        moves = picking.move_lines
        for move in moves:
```